compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -599,6 +599,17 @@ impl<'a> Parser<'a> {
         self.last_unexpected_token_span = Some(self.token.span);
         let mut err = self.struct_span_err(self.token.span, &msg_exp);
 
+        if let TokenKind::Ident(symbol, _) = &self.prev_token.kind {
+            if symbol.to_string() == "public" {
```

Comment:
I think it would be better if you added `public` to `rustc_span/src/symbol.rs` and then just compared the symbols here. I believe that should be faster.

Reply:
Does this case only occur on a parse error? I don't think we need to worry about performance in that case.

Reply:
That's fair enough, feel free to do this or not when you're adding tests.

Reply:
There's an `Symbol::as_str` which might be preferred here.